### ISSUE TITLE
Add PropTypes in all react components for static typechecking

### ISSUE THREAD

**Issue by OscarWang114** [wasedatime/wasedatime-web#24]
### Notes
Please use the git branch **`prop-types`** for developing.
By resolving this issue, you can gain good familiarity with the React components in wasetime-web 😄. Please read through this [React doc](https://reactjs.org/docs/typechecking-with-proptypes.html) for details and good examples. 
### Issue
`PropTypes` is useful for static checking and reducing bugs in the code, but it is not fully used in this project. **We need to add PropTypes for all React components to prevent potential bugs.**

For example, in `Root.js` we use `PropTypes` as follows: 
```javascript
Root.propTypes = {
  store: PropTypes.object.isRequired
};
```
This ensures that:
1. `store` is always passed on to the component `Root` (because of `.isRequired`)
2. The type of `store` is an object (because of `.object`)

**We need to do so for every React component in this project.**

### Example
`AddedCourseItem` does not have `PropTypes` checking yet. So we should check its props by:
**1. Checking what kind of props is passed by its parent component.** (here it is `AddedCourseList`). In `AddedCourseList.js` we have:
```javascript
<AddedCourseItem key={course._id} course={course} />
```
Note that `key` is a **special props used by React and is NOT passed on to the children component ([Related React doc](https://reactjs.org/warnings/special-props.html))** Therefore, we know that `AddedCourseItem` has props `course` (type `object`)

**2. Checking the component itself.** In `AddedCourseItem.js` we have:
```
const mapDispatchToProps = {
  removeCourse
};

export default connect(null, mapDispatchToProps)(AddedCourseItem);
```
Therefore, we know that `AddedCourseItem` has props `removeCourse` (type `function`)

Combining these two points we know we need to add the followings at the end of `AddedCourseItem.js`:
```javascript
AddedCourseItem.propTypes = {
  course: PropTypes.object.isRequired,
  removeCourse: PropTypes.func.isRequired
};
```
And we are done! You can refer to the actual code [here](https://github.com/wasetime/wasetime-web/commit/9ab4a20dc467a98406f911d5b6ca2bfdaf473b84).

**Comment by YHhaoareyou** [comment]
I would like to work on this issue.

**Comment by OscarWang114** [comment]
@YHhaoareyou Great! I look forward to your pull request 😄. Feel free to contact me if you have any questions!

**Comment by bill291212** [comment]
I would like to work on this issue

**Comment by OscarWang114** [comment]
@bill291212 Gotcha! Looking forward to yoru pull request as well 😃. Let me know if you have any questions!

**Comment by OscarWang114** [comment]
Resolved by #26 . Closing.
